python/lsst/ts/adamSensors/adamSensorsCSC.py

```diff
@@ -44,9 +40,23 @@ async def begin_start(self, data):
         self.cmd_start.ack_in_progress(data, timeout=self.start_timeout)
         await super().begin_start(data)
         self.adam = AdamModel(self.log, simulation_mode=self.simulation_mode)
-        self.adam.connect(self.params.ip, self.params.port)
+        try:
+            await self.adam.connect(self.config.adam_ip, self.config.adam_port)
+        except ConnectionException:
+            self.fault(code=2, report="Unable to connect to modbus device at " + str(self.config.adam_ip))
+        if self.telemetry_loop_task.result() is not None:
+            self.telemetry_loop_task.cancel()
         self.telemetry_loop_task = asyncio.create_task(self.telemetry_loop())
 
+    async def end_standby(self, data):
+        """
+        When transitioning from disabled or fault state to standby,
+        cancels the telemetry loop task and disconnects from the ADAM
+        device.
+        """
+        self.telemetry_loop_task.cancel()
+        await self.model.disconnect()
```

Comment:
you may want to add a try/except clause here to prevent the command from failing if this line fails. Otherwise you may be left unable to transition back to standby. You may also want to wait for `self.telemetry_loop_task`

Reply:
Unfortunately i don't have access to the hardware now so I can't figure out what the exception would be by experimenting. Do you know what exception i'd be catching in this case? 

Also do you mean do you mean by "wait for" `self.telemetry_loop_task`? like `await` it?

Reply:
Basically you want to do something like this:

```
if not self.telemetry_loop_task.done(): 
    self.telemetry_loop_task.cancel()

try:
    await self.telemetry_loop_task
except asyncio.CancelledError:
    pass
except Exception:
    self.log.exception("Exception in telemetry loop.")

try:
    await self.model.disconnect()
except Exception:
    self.log.exception("Error disconnecting from controller." )
```